test/gtest/uct/uct_test.cc

```diff
@@ -1081,24 +1086,70 @@ void uct_test::entity::destroy_eps() {
     }
 }
 
-size_t uct_test::entity::priv_data_do_pack(void *priv_data)
+static void set_data(int *initialized, std::string& data, size_t len)
 {
-    size_t priv_data_len;
+    if (!(*initialized)) {
+        data.resize(len);
+        ucs::fill_random(data);
+        (*initialized) = 1;
+    }
+}
 
-    client_priv_data = "Client private data";
-    priv_data_len = 1 + client_priv_data.length();
+size_t uct_test::entity::priv_data_do_pack(void *priv_data, size_t pack_limit)
+{
+    static int client_data_initialized;
 
+    if (pack_limit < long_data_len) {
+        if (priv_data_len != short_data_len) {
+            client_data_initialized = 0;
+        }
+        /* non-cm flow should get here, as well as cm flow with a transport
+         * with a small private data length (rdmacm_cm) */
+        priv_data_len = short_data_len;
+    } else {
+        if (priv_data_len != long_data_len) {
+            client_data_initialized = 0;
+        }
+        priv_data_len = long_data_len;
+    }
```

Comment:
lets define one class for all priv_data sizes with interface like:
```
static class {
public:
    void *get(size_t length);
...
} sa_priv_data;
```
or maybe return `std::string` instead of `void *` if we use strings for comparison everywhere

Reply:
@evgeny-leksikov which logic do you mean to replace here? 
please note that class test_uct_cm_sockaddr will have diff pack_limits for tcp_sockcm and for rdmacm

Reply:
> @evgeny-leksikov which logic do you mean to replace here?
> please note that class test_uct_cm_sockaddr will have diff pack_limits for tcp_sockcm and for rdmacm

all this from line 1102 to 1114 + definitions here https://github.com/openucx/ucx/pull/5073#discussion_r420641431